exporters/dynamicconfig/confignotifier.go

```diff
@@ -0,0 +1,99 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dynamicconfig
+
+import (
+	"fmt"
+	"time"
+)
+
+type MetricConfig struct {
+	Period time.Duration
+}
+
+func (config *MetricConfig) equals(otherConfig *MetricConfig) bool {
+	if config == nil && otherConfig == nil {
+		return true
+	}
+
+	if config == nil || otherConfig == nil {
+		return false
+	}
+
+	configString := fmt.Sprintf("%+v", *config)
+	otherConfigString := fmt.Sprintf("%+v", *otherConfig)
+
+	if configString == otherConfigString {
+		return true
+	}
+	return false
+}
```

Comment:
Could we simplify this logic to:
```go
// make config strings

isEqual := configString == otherConfigString
return isEqual
```
If both `configString` and `otherConfigString` are `nil`, would `fmt.Sprintf` still yield the same string for both at the end?

Reply:
What are your thoughts on `cmp.Equal`, which may be more rigorous than converting to strings?

Reply:
I separate the config equality logic into it's own function because this entire function body is probably going to end up replaced in the end. The reason we can do a simple string comparison right now is because the config only has one field.  In the future, when we have a better idea of what a config would look like, this function body will probably do some sort of deep comparison.

I'm not sure about `cmp.Equal`. Apparently it's not very performant and mostly meant for testing, which might an issue if the config is sufficiently big, and we check a remote service sufficiently often (https://godoc.org/github.com/google/go-cmp/cmp).